pkg/blackduck/ctl_blackduck.go

```diff
@@ -443,71 +313,32 @@ func (ctl *CRSpecBuilderFromCobraFlags) SetCRSpecFieldByFlag(f *pflag.Flag) {
 				log.Errorf("failed to unmarshal security contexts: %+v", err)
 				return
 			}
-			ctl.blackDuckSpec.SecurityContexts = SecurityContexts
-		case "postgres-claim-size":
-			for i := range ctl.blackDuckSpec.PVC {
-				if ctl.blackDuckSpec.PVC[i].Name == "blackduck-postgres" { // update claim size and return
-					ctl.blackDuckSpec.PVC[i].Size = ctl.PostgresClaimSize
-					return
-				}
-			}
-			ctl.blackDuckSpec.PVC = append(ctl.blackDuckSpec.PVC, blackduckv1.PVC{Name: "blackduck-postgres", Size: ctl.PostgresClaimSize}) // add postgres PVC if doesn't exist
-		case "certificate-name":
-			ctl.blackDuckSpec.CertificateName = ctl.CertificateName
-		case "certificate-file-path":
-			data, err := util.ReadFileData(ctl.CertificateFilePath)
-			if err != nil {
-				log.Fatalf("failed to read certificate file: %+v", err)
-			}
-			ctl.blackDuckSpec.Certificate = data
-		case "certificate-key-file-path":
-			data, err := util.ReadFileData(ctl.CertificateKeyFilePath)
-			if err != nil {
-				log.Fatalf("failed to read certificate file: %+v", err)
-			}
-			ctl.blackDuckSpec.CertificateKey = data
-		case "proxy-certificate-file-path":
-			data, err := util.ReadFileData(ctl.ProxyCertificateFilePath)
-			if err != nil {
-				log.Fatalf("failed to read certificate file: %+v", err)
+			for k, v := range SecurityContexts {
+				util.SetHelmValueInMap(ctl.args, []string{k, "securityContext"}, OperatorSecurityContextTok8sAffinity(v))
 			}
-			ctl.blackDuckSpec.ProxyCertificate = data
-		case "auth-custom-ca-file-path":
-			data, err := util.ReadFileData(ctl.AuthCustomCAFilePath)
-			if err != nil {
-				log.Fatalf("failed to read authCustomCA file: %+v", err)
-			}
-			ctl.blackDuckSpec.AuthCustomCA = data
-		case "type":
-			ctl.blackDuckSpec.Type = ctl.Type
+		case "postgres-claim-size":
+			util.SetHelmValueInMap(ctl.args, []string{"postgres", "claimSize"}, ctl.PostgresClaimSize)
+
 		case "desired-state":
-			ctl.blackDuckSpec.DesiredState = ctl.DesiredState
+			util.SetHelmValueInMap(ctl.args, []string{"status"}, ctl.DesiredState)
 		case "migration-mode":
-			if ctl.MigrationMode {
-				ctl.blackDuckSpec.DesiredState = "DbMigrate"
-			}
-		case "image-registries":
-			ctl.blackDuckSpec.ImageRegistries = ctl.ImageRegistries
-		case "license-key":
-			ctl.blackDuckSpec.LicenseKey = ctl.LicenseKey
+			util.SetHelmValueInMap(ctl.args, []string{"status"}, "DbMigrate")
```

Comment:
I don't think the BlackDuck Helm Chart can handle DbMigrate right now

Reply:
@msenmurugan Is there any plan to add this feature or should we just get rid of this?

Reply:
there is already a stop status, which should be enough.